src/lib/alloc.c

```diff
@@ -70,6 +62,32 @@ extern struct mm memmap;
  *    module removal or calls to rfree(). Saved as part of PM context.
  */
 
+#if DEBUG_BLOCK_FREE
+/* Check whole memory region for debug pattern to find if memory was feed
+ * second time
+ */
+static void validate_memory(void *ptr, size_t size)
+{
+	uint8_t *ptr_8 = ptr;
+	int i, not_matching = 0;
+
+	for (i = 0; i < size; i++) {
+		if (ptr_8[i] != DEBUG_BLOCK_FREE_VALUE)
```

Comment:
I think our smallest unit for allocation and alignment is cache line size, so we should be searching for a uint32_t on a cache line boundary with 0xa5a5a5a5

Reply:
> I think our smallest unit for allocation and alignment is cache line size

That is true for system heap, for other allocs we do not align to cashe line. 
And this feature doesn't support system heap.

Left it as 1 byte to align with memset() which is used to write the pattern
and not to create additional defines.

Reply:
Changed